### PR TITLE
BH-311: Supporting documents tab

### PR DESCRIPTION
**Pull request by TimLeach635**
## Things to note
* There was a graphic on the Figma page that looks like this:
  image
  I implemented it as a `.svg` embedded directly on the page, and did my best to give a descriptive `alt` for the ``
* The banner only disappears on page refresh, so if the user switches over to the "supporting documents" tab and then switches back to the "referral files" tab, the banner will still be there until refresh. I think this is fine!
* On Figma, the banner itself has a comment from Kay saying it didn't test very well (half of participants just ignored it), so I have a feeling this whole design will change!

## Banner
On first login:
image
After refresh:
image

### CODE REVIEW

File: HerPortal.Data/IDataAccessProvider.cs [hunk 1]
```diff
@@ -1,5 +1,9 @@
-﻿namespace HerPortal.Data;
+﻿using HerPortal.BusinessLogic.Models;
+
+namespace HerPortal.Data;
```
Comment (DanCorderSoftwire) [inline HerPortal.Data/IDataAccessProvider.cs:3]: Can we move this file to the business logic project please?

File: HerPortal.Data/Migrations/HerDbContextModelSnapshot.cs [hunk 2]
```diff
@@ -21,6 +21,59 @@ protected override void BuildModel(ModelBuilder modelBuilder)
...
+
+                    b.Property<bool>("HasLoggedIn")
+                        .HasColumnType("boolean");
+
+                    b.HasKey("Id");
+
+                    b.ToTable("Users");
+                });
+
+            modelBuilder.Entity("LocalAuthorityUser", b =>
```
Comment (DanCorderSoftwire) [inline HerPortal.Data/Migrations/HerDbContextModelSnapshot.cs:62]: I don't think you need this if you specify something like this on either the local authority or user tables:

```
 modelBuilder.Entity<LocalAuthority>()
        .HasMany(e => e.Users)
        .WithMany(e => e.LocalAuthorities);
```

https://learn.microsoft.com/en-us/ef/core/modeling/relationships/many-to-many

File: HerPortal.Data/Migrations/HerDbContextModelSnapshot.cs [hunk 3]
```diff
@@ -39,6 +92,21 @@ protected override void BuildModel(ModelBuilder modelBuilder)
 
                     b.ToTable("DataProtectionKeys");
                 });
+
+            modelBuilder.Entity("LocalAuthorityUser", b =>
```
Comment (DanCorderSoftwire) [inline HerPortal.Data/Migrations/HerDbContextModelSnapshot.cs:96]: Don't think you need this either

File: HerPortal/DataStores/UserDataStore.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
+﻿using System.Threading.Tasks;
```
Comment (DanCorderSoftwire) [inline HerPortal/DataStores/UserDataStore.cs:1]: I'm not sure what this class is doing for us? Will it be more useful later on?

File: HerPortal/Models/HomepageViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,13 @@
+﻿using HerPortal.BusinessLogic.Models;
+
+namespace HerPortal.Models;
+
+public class HomepageViewModel
+{
+    public bool ShouldShowBanner { get; }
+
+    public HomepageViewModel(User user)
```
Comment (DanCorderSoftwire) [inline HerPortal/Models/HomepageViewModel.cs:9]: This is fairly trivial at the moment, but it's still a good candidate for a unit test, to get that kicked off

**Review by DanCorderSoftwire — CHANGES_REQUESTED** [review]

File: HerPortal.Data/Migrations/HerDbContextModelSnapshot.cs (hunk 2 above)
Comment (TimLeach635) [reply HerPortal.Data/Migrations/HerDbContextModelSnapshot.cs:62]: The `DbContextModelSnapshot` is auto-generated, I use [Entity Framework conventions](https://www.entityframeworktutorial.net/code-first/configure-many-to-many-relationship-in-code-first.aspx) (also see [here](`BusinessLogic.Models.User`)) in `BusinessLogic.Models.User` and `BusinessLogic.Models.LocalAuthority` to replicate what you've done there with Fluent assertions 😊 

Even specifying that, EF still needs to generate the join table, so it shows up here!

File: HerPortal.Data/Migrations/HerDbContextModelSnapshot.cs (hunk 3 above)
Comment (TimLeach635) [reply HerPortal.Data/Migrations/HerDbContextModelSnapshot.cs:96]: As above, this file is auto-generated

File: HerPortal.Data/Migrations/HerDbContextModelSnapshot.cs (hunk 2 above)
Comment (DanCorderSoftwire) [reply HerPortal.Data/Migrations/HerDbContextModelSnapshot.cs:62]: Sorry, my mistake, I thought I was looking at a different file!

File: HerPortal/DataStores/UserDataStore.cs (hunk 4 above)
Comment (DanCorderSoftwire) [reply HerPortal/DataStores/UserDataStore.cs:1]: In this case I think so. I don't think we're planning to do anything complicated with users so this class doesn't seem to be adding anything. Apologies in advance if we decide to add it back later on :)

File: HerPortal.UnitTests/Website/Models/HomepageViewModelTests.cs [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+using HerPortal.Models;
+using NUnit.Framework;
+
+namespace Tests.Website.Models;
+
+[TestFixture]
+public class HomepageViewModelTests
+{
+    [Test]
+    public void ShouldShowBannerIfUserHasNotLoggedIn()
```
Comment (DanCorderSoftwire) [inline HerPortal.UnitTests/Website/Models/HomepageViewModelTests.cs:12]: We're trying to go with a test naming pattern along the lines of `HomePageViewModel_WhenCreatedForUserThatHasntLoggedInBefore_ShouldShowBanner` i.e. `TheThingOrMethodBeingTested_TheCase_TheResult`

File: HerPortal.UnitTests/Website/Models/HomepageViewModelTests.cs [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        
+        // Act
+        var viewModel = new HomepageViewModel(user);
+        
+        // Assert
+        viewModel.ShouldShowBanner.Should().BeTrue();
+    }
+    
+    [Test]
+    public void ShouldNotShowBannerIfUserHasLoggedIn()
```
Comment (DanCorderSoftwire) [inline HerPortal.UnitTests/Website/Models/HomepageViewModelTests.cs:28]: Two tests is fine, but be aware that you could also write this as one test with two [TestCase(true, false)] annotations

**Review by DanCorderSoftwire — APPROVED** [review]

**Pull request merged**